Extensions: Incremental additions

maryamariyan opened the pull request:
**This PR:**
- [x] ref/src/pkg (have no test projects)
    - Microsoft.Extensions.Caching.Abstractions
    - Microsoft.Extensions.Configuration.Abstractions
    - Microsoft.Extensions.DependencyInjection.Abstractions
    + Microsoft.Extensions.Hosting.Abstractions
    - Microsoft.Extensions.Logging.Abstractions

- [x] ref/src/test/pkg
    - Microsoft.Extensions.Configuration
    - Microsoft.Extensions.Configuration.Binder
    - Microsoft.Extensions.Configuration.CommandLine
    - Microsoft.Extensions.Configuration.EnvironmentVariables
    - Microsoft.Extensions.Options.ConfigurationExtensions
    - Microsoft.Extensions.Options.DataAnnotations
----
**Next PRs:**

Disabled in Directory.Build.props until pkg/test gets completed.
Could defer this to a next PR.
- [x] ref/src/pkg
- [x] test (reason for later: https://github.com/dotnet/runtime/pull/33563#issuecomment-598981289)
    - Microsoft.Extensions.DependencyInjection
    - Microsoft.Extensions.Options (Options tests needs ref to DI)


- [x] ref/src/pkg/test
    - Microsoft.Extensions.Caching.Memory 
    - Microsoft.Extensions.Configuration.FileExtensions
    - Microsoft.Extensions.Configuration.Ini
    - Microsoft.Extensions.Configuration.Json
    - Microsoft.Extensions.Configuration.UserSecrets
    - Microsoft.Extensions.Configuration.Xml
    - Microsoft.Extensions.Hosting
    - Microsoft.Extensions.Http
    - Microsoft.Extensions.Logging.* (8x)

- [ ] Enable tests explained in comment: https://github.com/dotnet/runtime/pull/33563#issuecomment-598968562
- [ ] Add solution files

maryamariyan committed:
Enables ref/src for:

- Microsoft.Extensions.Caching.Abstractions
- Microsoft.Extensions.Configuration
- Microsoft.Extensions.Configuration.Abstractions
- Microsoft.Extensions.Configuration.Binder
- Microsoft.Extensions.Configuration.CommandLine
- Microsoft.Extensions.DependencyInjection
- Microsoft.Extensions.DependencyInjection.Abstractions
- Microsoft.Extensions.Logging.Abstractions
- Microsoft.Extensions.Options

maryamariyan committed:
Preliminary cleanup for:
- Microsoft.Extensions.Caching.Abstractions
- Microsoft.Extensions.Caching.Memory
- Microsoft.Extensions.Http

- TODO
[ ] Add Logging before Http and Caching.Memory

maryamariyan committed:
ref/src
+ Microsoft.Extensions.DependencyInjection
+ Microsoft.Extensions.Hosting.Abstractions

test
+ Microsoft.Extensions.Configuration.Tests
+ Microsoft.Extensions.Configuration.Binder.Tests
+ Microsoft.Extensions.Configuration.CommandLine.Tests

maryamariyan committed:
ref/src
+ Microsoft.Extensions.Options.ConfigurationExtensions

maryamariyan committed:
+ pkg

- Microsoft.Extensions.Caching.Abstractions
- Microsoft.Extensions.Configuration.Abstractions
- Microsoft.Extensions.DependencyInjection.Abstractions
- Microsoft.Extensions.Logging.Abstractions
- Microsoft.Extensions.Hosting.Abstractions

ericstj commented:
```
src/libraries/Microsoft.Extensions.DependencyInjection/tests/Common/test/Microsoft.Extensions.Logging.Testing.Tests.csproj#L2
src/libraries/Microsoft.Extensions.DependencyInjection/tests/Common/test/Microsoft.Extensions.Logging.Testing.Tests.csproj(2,3): error MSB4019: The imported project "/Users/runner/runners/2.165.0/work/1/s/src/libraries/Microsoft.Extensions.DependencyInjection/tests/Logging.Testing/src/build/Microsoft.Extensions.Logging.Testing.props" was not found. Confirm that the expression in the Import declaration "/Users/runner/runners/2.165.0/work/1/s/src/libraries/Microsoft.Extensions.DependencyInjection/tests/Common/test/../../Logging.Testing/src/build/Microsoft.Extensions.Logging.Testing.props" is correct, and that the file exists on disk
```

Looks like we need to update the path in src/libraries/Microsoft.Extensions.DependencyInjection/tests/Common/test/Microsoft.Extensions.Logging.Testing.Tests.csproj

maryamariyan committed:
ref/src
+ Microsoft.Extensions.Options.DataAnnotations

maryamariyan committed:
+ pkg

- Microsoft.Extensions.Configuration
- Microsoft.Extensions.Configuration.Binder
- Microsoft.Extensions.Configuration.CommandLine

maryamariyan committed:
CI check: defer pkg/test on Options and DependencyInjection

maryamariyan committed:
+ pkg

- Microsoft.Extensions.DependencyInjection
- Microsoft.Extensions.Options
- Microsoft.Extensions.Options.ConfigurationExtensions
- Microsoft.Extensions.Options.DataAnnotations

maryamariyan committed:
leftover GenerateReferenceSource target changes

maryamariyan committed:
+ test
-Microsoft.Extensions.Options

maryamariyan commented:
The following 6 Microsoft.Extensions test projects (under src/libraries) are not yet cleaned up from when they were brought over from Extensions repo:
```
Microsoft.Extensions.Configuration\tests\FunctionalTests\Microsoft.Extensions.Configuration.Functional.Tests.csproj
Microsoft.Extensions.DependencyInjection\tests\Common\test\Microsoft.Extensions.Logging.Testing.Tests.csproj
Microsoft.Extensions.Hosting\tests\FunctionalTests\Microsoft.Extensions.Hosting.Functional.Tests.csproj
Microsoft.Extensions.Hosting\tests\TestApp\Microsoft.Extensions.Hosting.TestApp.csproj
Microsoft.Extensions.Hosting\tests\UnitTests\Microsoft.Extensions.Hosting.Unit.Tests.csproj
Microsoft.Extensions.Logging\tests\Common\Microsoft.Extensions.Logging.Tests.csproj
```
They can't be enabled yet, until all their dependencies are brought over. I'll enable them once the dependencies are completely enabled:

- First one needs to get enabled once all Configurations assemblies are in. (not yet)
- Second one needs Logging, and in fact should be moved under Logging test folder not DI
- The remaining 4, need Hosting and Logging need to be present first.
----
To keep the remaining test projects enabled I modified 
```
<ProjectExclusions Include="$(MSBuildThisFileDirectory)*\tests\**\Microsoft.Extensions.*.Tests.csproj" Exclude="@(PortedExtensionsProject)" />
```
to
```
<ProjectExclusions Include="$(MSBuildThisFileDirectory)*\tests\Microsoft.Extensions.*.Tests.csproj" Exclude="@(PortedExtensionsProject)" />
```
in the Directory.Build.props file

maryamariyan commented:
- [ ] To enable Microsoft.Extensions.DependencyInjection.Tests I need to bring over history for [DI.Specification.Tests](https://github.com/dotnet/extensions/tree/master/src/DependencyInjection/DI.Specification.Tests/src) as seen in the csproj below:

https://github.com/dotnet/runtime/blob/master/src/libraries/Microsoft.Extensions.DependencyInjection/tests/Microsoft.Extensions.DependencyInjection.Tests.csproj#L10

maryamariyan committed:
+ ref/src/pkg/test
Microsoft.Extensions.Configuration.EnvironmentVariables

maryamariyan committed:
test project exclusions

maryamariyan committed:
DIsable DI, its test needs DI.Specifactions not in repo yet
DIsable Options, its test needs DI
Cleanup ProjectExclusions to skip csproj in Common folder

ericstj commented on src/libraries/Microsoft.Extensions.Caching.Abstractions/src/Microsoft.Extensions.Caching.Abstractions.csproj:
```diff
@@ -1,22 +1,13 @@
   </PropertyGroup>
 
   <ItemGroup>
+    <Compile Include="**/*.cs" />
```
I may have missed this in past PRs.  In general we try to avoid the wildcards in the CSProj's but that's just because we condition imports.  There are other ways to do this that might be preferable.  Maybe we set `EnableDefaultItems`  @ViktorHofer any opinions on this one?

ViktorHofer replied:
Yes, what Eric said. If we don't plan to multi-target for different OSs then I would use `EnableDefaultItems` here.

maryamariyan committed:
using EnableDefaultItems

maryamariyan commented on src/libraries/Microsoft.Extensions.Configuration/tests/Microsoft.Extensions.Configuration.Tests.csproj:
```diff
@@ -5,13 +5,22 @@
   </ItemGroup>
 
   <ItemGroup>
     <Reference Include="Microsoft.Extensions.Configuration.Abstractions" />
```
I'm following the same pattern as in [System.ComponentModel.TypeConverter.Tests](https://github.com/dotnet/runtime/blob/master/src/libraries/System.ComponentModel.TypeConverter/tests/System.ComponentModel.TypeConverter.Tests.csproj#L161)

Some of these test projects need Moq

ericstj replied:
I mean delete all `<Reference` items.  Tests don't need these, by default tests reference everything we build.

maryamariyan commented on src/libraries/Microsoft.Extensions.Configuration.CommandLine/tests/Microsoft.Extensions.Configuration.CommandLine.Tests.csproj:
```diff
@@ -5,14 +5,33 @@
+  <ItemGroup>
+    <!-- Some internal types are needed, so we reference the implementation assembly, rather than the reference assembly. -->
+    <DefaultReferenceExclusions Include="Microsoft.Extensions.Configuration.CommandLine" />
+    <ReferenceFromRuntime Include="Microsoft.Extensions.Configuration.CommandLine" />
```
Also here, I followed the workflow in other projects like [System.Numerics.Tensors.Tests.csproj](https://github.com/dotnet/runtime/blob/master/src/libraries/System.Numerics.Tensors/tests/System.Numerics.Tensors.Tests.csproj#L36-L39)

ericstj replied:
I would have expected you to only need ReferenceFromRuntime, but I do see this duplication in a lot of projects.  If you tried omitting DefaultReferenceExclusions and it didn't work then I'm fine leaving it as is.  We should cleanup the whole repo with a follow up issue that moves this duplication into the [targets](https://github.com/dotnet/runtime/blob/master/eng/referenceFromRuntime.targets).

maryamariyan committed:
using GenerateReferenceSource
- used DefaultApiDocIds to persist ObsoleteAttribute

maryamariyan committed:
Removes references on test projects

maryamariyan committed:
Fixup SR.Format arguments

maryamariyan committed:
Test csproj cleanup:
- Remove DefaultReferenceExclusions and more

maryamariyan commented:
(seems unrelated but) noticed below issue in the CI: https://dev.azure.com/dnceng/public/_build/results?buildId=561783
```
XUnit : error : Tests failed: /root/runtime/artifacts/TestResults/Debug/AppHost.Bundle.Tests_netcoreapp5.0_x64.html [netcoreapp5.0|x64] [/root/runtime/src/installer/test/Microsoft.NET.HostModel.Tests/AppHost.Bundle.Tests/AppHost.Bundle.Tests.csproj]
##[error]XUnit(0,0): error : (NETCORE_ENGINEERING_TELEMETRY=Test) Tests failed: /root/runtime/artifacts/TestResults/Debug/AppHost.Bundle.Tests_netcoreapp5.0_x64.html [netcoreapp5.0|x64]
```

ericstj commented on src/libraries/Microsoft.Extensions.FileProviders.Physical/tests/Microsoft.Extensions.FileProviders.Physical.Tests.csproj:
```diff
@@ -16,20 +16,7 @@
   </ItemGroup>
 
   <ItemGroup>
-    <PackageReference Include="Moq" Version="$(MoqVersion)" />
```
You still need `<PackageReference Include="Moq" Version="$(MoqVersion)" />`, sorry if my previous comments around `<Reference` were not clear since I was commenting on the unchanged lines.

maryamariyan committed:
CI fix - Bring back Moq for FileProviders.Physical.Tests
